ContactList/ContactList/TableViewController.swift

```diff
@@ -32,125 +37,221 @@ class TableViewController: UITableViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         tableView.backgroundView = emptyListView
-        buttonCopy = addNewContactButton
-        personsArray = WorkWithData.getArray()
-    }
-    
-    func addContactButtonSetVisibility() {
-        navigationItem.setRightBarButton(!personsArray.isEmpty ? buttonCopy : nil, animated: true)
+//        buttonCopy = addNewContactButton
+        buttonsCopy = (editTableButton, addNewContactButton)
+        groupedPersons = DataManager.getDictionary()
+        createSearchBar()
+        changeSearchBarVisibility()
     }
 
     // MARK: - Table view data source
-
     override func numberOfSections(in tableView: UITableView) -> Int {
-        
-        addContactButtonSetVisibility()
-        
-        if personsArray.isEmpty {
+        changeButtonsVisibility()
+        if groupedPersons.isEmpty {
```

Comment:
Ok for now, but in next iteration move code that hides `tableView.backgroundView` to 
```
private (set) var groupedPersons: [String : [Person]]! {
    didSet {
        ...
    }
}
```